### ISSUE TITLE
the original prefix-tuning corresponds to the models in the directory `models/prompt`?

### ISSUE THREAD

**Issue by Chacha-Chen** [HKUNLP/UnifiedSKG#34]
Dear authors,

Thank you so much for the effort and opensourcing the well-maintained and clean codebase! And I also appreciate your detailed explanations on Zhihu. 

I just have a quick specific question, are the scripts inside `models/prompt` the re-implementation of the prefix tuning paper (Li & Liang)? 

Thank you so much!

**Comment by Chacha-Chen** [comment]
I guess my questions are more like "how to run the basic prefix tuning version? what is the parameters/config that I need to change?

Thanks for your answer in advance!

**Comment by Chacha-Chen** [comment]
Oh. I think I may have found the answer. If I understand correctly, the single task prefix tuning is using t5_xxx_prefix_{task}.cfg.

**Comment by Chacha-Chen** [comment]
But how to run the code if I don't want to use any knowledge? simply setting `knowledge_usage` to None will have error for multiwoz data

**Comment by Timothyxxx** [comment]
Hi,

Thanks for your question!
What is your config file looks like?

Thanks

**Comment by Timothyxxx** [comment]
Oh I see the problem.

If you set the `knowledge_usage`(means the represent of `query`, `context` and `structured knwoledge`) to "None", different from not setting it(which will make it `None` and be defaulted to use the `concatenate` way of construct `seq_in` in [here](https://github.com/HKUNLP/UnifiedSKG/blob/49a2ff950bb312b980c22ad72b11520db72ab6a3/utils/dataset.py#L30)). 
Then in our implementation for now, it will raise ValueError.
In this case, I recommend to create a new loader under `seq2seq_construction` file, and add a arg like `args.seq2seq.use_apis` like what we did in [here](https://github.com/HKUNLP/UnifiedSKG/blob/main/seq2seq_construction/mtop.py) to set the structured knowledge to "".

Contact me if you have more problems, btw, nice to meet you in both Zhihu and here!

**Comment by Chacha-Chen** [comment]
Thanks for the prompt response! 
Just as a sanity check, if I do what you suggested, I will be running the basic version of prefix tuning (Li & Liang) right? 
I am asking cause I feel `knowledge_usage` thing is from your paper but not sure. And also should I make changes to `use_description` and `concatenate_description` as well? 

Thank you again for the help!

**Comment by Timothyxxx** [comment]
Yes, the `knowledge_usage`, `use_description` and `concatenate_description` are all from our own attempts. Among them, if you set the `knowledge_usage` to 'concatenate' or choose to not set it, it will use the basic version of Prefix-tuning, but if you use set it to the 'separate', it will be our changes on the prefix passing mechanism(but we found not work.). Args `use_description` and its value arg `concatenate_description` and `map_description` only useful for multi-task settings, `concatenate_description` means use description as the prefix of the seq_in while `map_description` means using description as a mapping controller(yeah, here the setting is different, we tried and also didn't work significantly better than basic one).
So if you run a single task, just don't set `knowledge_usage ` and set `use_description` to False, and write the loader with struct_in set to "" will be okay.
